firebase-storage/src/test/java/com/google/firebase/storage/TestUtil.java

```diff
@@ -32,6 +32,12 @@
 public class TestUtil {
 
   static FirebaseApp createApp() {
+    // Many tests require you to call the callback on the same thread that was initially
+    // instantiated.
+    // With the 5 second keepalive, after 5 seconds, the thread will get killed and eventually a new
+    // one will be created.
+    // Therefore causing many of the tests to fail
+    StorageTaskScheduler.setCallbackQueueKeepAlive(90, TimeUnit.SECONDS);
```

Comment:
To add additional context:

We call `attachListeners` https://github.com/firebase/firebase-android-sdk/blob/ed10eb5ebe998dc23ead46419a2b4dcbb5e2482c/firebase-storage/src/testUtil/java/com/google/firebase/storage/TestUploadHelper.java#L49 to set up the listeners and then those listeners call `ControllableSchedulerHelper` to verify the callback thread that it initially was called with is the same one that is called in subsequent events https://github.com/firebase/firebase-android-sdk/blob/ed10eb5ebe998dc23ead46419a2b4dcbb5e2482c/firebase-storage/src/testUtil/java/com/google/firebase/storage/ControllableSchedulerHelper.java#L63
However, in cases where we have long backoffs, this won't be the case, as the thread pool will terminate any threads that are not executed within the keepalive time, which is 5 seconds.
Our alternative is to set the keepAlive, or disable the threads from being killed. A third option would be to remove this verification check.

Reply:
I think in case the wrong executor gets called by `UploadTask`. 

Here's a quick StackOverflow answer that talks about the tradeoffs:
https://stackoverflow.com/a/18225863